dotCMS/src/main/java/com/dotcms/rest/api/v1/site/SiteHelper.java

```diff
@@ -159,23 +116,94 @@ public Host getSite(User user, String siteId) throws DotSecurityException, DotDa
 	 * @param siteInSession
 	 *            - The Identifier of the site that is marked as selected in the
 	 *            current user session.
+	 * @param user
+	 *            - The current user session.            
 	 * @return The Identifier of the site that will be marked as "selected".
+	 * @throws DotSecurityException 
+	 * @throws DotDataException 
 	 */
-	public String getSelectedSite(final List<Map<String, Object>> siteList, final String siteInSession) {
+    public String getSelectedSite(final List<Host> siteList, final String siteInSession, User user)  {
 		String selectedSite = UtilMethods.isSet(siteInSession) ? siteInSession : StringUtils.EMPTY;
 		boolean siteFound = false;
 		if (siteList != null && !siteList.isEmpty()) {
-			for (Map<String, Object> siteData : siteList) {
-				if (siteData.get(Contentlet.IDENTIFIER_KEY).equals(siteInSession)) {
-					siteFound = true;
-					break;
+			Host host = null;
+			try {
+				host = this.hostAPI.find(siteInSession, user, Boolean.FALSE);
+			} catch (DotDataException e) {
+				/** The host can't be found **/
+			} catch (DotSecurityException e) {
+				/** The user doesn't have permission to see this host **/
+			}
+			
+			if (null != host && UtilMethods.isSet(host.getIdentifier())) {
+				siteFound = true;
+				if(!siteList.contains(host)){
```

Comment:
That is a list because the siteList is the results of the PaginatedArrayList I'm receiving a to get the total number of site the user have that list only contains 1 element. the cost of the contains is not a problem here the paginatedarrey list here alway receive only one element plus the counter of the total site the user have permissions